packaging/greenboot/microshift-running-check.sh

```diff
@@ -0,0 +1,239 @@
+#!/bin/bash
+set -e
+
+SCRIPT_NAME=$(basename $0)
+SCRIPT_PID=$$
+OCGET_CMD="microshift get"
+OCGET_OPT="--kubeconfig /var/lib/microshift/resources/kubeadmin/kubeconfig --no-headers"
+PODS_NS_LIST=(openshift-ovn-kubernetes openshift-service-ca openshift-ingress openshift-dns openshift-storage)
+PODS_CT_LIST=(2                        1                    1                 2             2)
+
+# Source Greenboot configuration file if it exists
+GREENBOOT_CONF_FILE=/etc/greenboot/greenboot.conf
+[ -f "${GREENBOOT_CONF_FILE}" ] && source ${GREENBOOT_CONF_FILE}
+WAIT_TIMEOUT_SECS_BASE=${MICROSHIFT_WAIT_TIMEOUT_SEC:-300}
+
+# Set the exit handler to log the exit status
+trap 'script_exit' EXIT
+
+# The script exit handler logging the FAILURE or FINISHED message depending
+# on the exit status of the last command
+#
+# args: None
+# return: None
+function script_exit() {
+    [ "$?" -ne 0 ] && status=FAILURE || status=FINISHED
+    echo $status
```

Comment:
How that works? Scripts sends `kill -TERM` for itself and this gets executed, greenboot makes a decision based on output `FAILURE` or `FINISHED`?

Reply:
The way bash EXIT handler works is that it's called on the script exit and "$?" contains the status of the last command. So, by adding a handler, we should always have FAILURE or SUCCESS printed in the end of the check, unless the script is killed with SIGKILL.

Reply:
Greenboot only checks the script exit code, never the output. This one is for user-friendly "visualization" of the result in the logs.

Reply:
And `kill -TERM` causes the script to exit with `1`?

Fine, I was kind of going over a incredible situation when we'd report status based on `$?` of echo but it's not an issue